lib/kubeclient/common.rb

```diff
@@ -405,6 +405,17 @@ def proxy_url(kind, name, port, namespace = '')
       end
     end
 
+    def process_template(template)
+      ns_prefix = build_namespace_prefix(template[:metadata][:namespace])
+      @headers['Content-Type'] = 'application/json'
+      response = handle_exception do
+        rest_client[ns_prefix + 'processedtemplates']
+        .post(template.to_h.to_json, @headers)
```

Comment:
Both indeed use ```Post``` on a template, but while ```create_template``` actually creates a template, ```process_template``` does not. It is only used to substitute the template parameters and generate their values (if required), in order to get a list of objects to create from the template. Note that we need here a separated method because although both requests are using ```Post``` they are still different: 

```POST /oapi/v1/namespaces/{namespace}/templates``` for creating a template. 

```POST /oapi/v1/namespaces/{namespace}/processedtemplates``` for processing a template.